enclave-runner/src/usercalls/mod.rs

```diff
@@ -346,36 +381,221 @@ impl EnclaveState {
             fds: Mutex::new(fds),
             last_fd,
             exiting: AtomicBool::new(false),
-            usercall_ext : usercall_ext,
+            usercall_ext,
         })
     }
 
+    fn create_worker_threads(num_of_worker_threads: usize,
+                             work_receiver: crossbeam::crossbeam_channel::Receiver<Work>,
+                             io_queue_send: mpsc::Sender<UsercallSendData>
+    ) -> Vec<JoinHandle<()>> {
+        let mut thread_handles = vec![];
+        for _ in 0..num_of_worker_threads {
+            let work_receiver = work_receiver.clone();
+            let io_queue_send = io_queue_send.clone();
+
+            thread_handles.push(thread::spawn(move || {
+                while let Ok(work) = work_receiver.recv() {
+                    work.do_work(&io_queue_send);
+                }
+            }));
+        }
+        return thread_handles;
+    }
+
+    fn  syscall_loop(io_queue_receive: mpsc::Receiver<UsercallSendData>, enclave: Arc<EnclaveState>,
```

Comment:
Instead of taking `enclave: Arc<EnclaveState>`, I think you can make this take `&self`. Same applies to `setup_and_start_work` below